edge-modules/MetricsCollector/MetricWithCustomTags.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Agent.Diagnostics
+{
+    using System;
+    using System.Collections.Generic;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public sealed class MetricWithCustomTags : Metric, IEquatable<Metric>
+    {
+        public IReadOnlyDictionary<string, string> CustomTags { get; }
+        public MetricWithCustomTags(Metric metric, IReadOnlyDictionary<string, string> customTags)
```

Comment:
why need Metric? you can directly pass TimeGeneratedUtc, Name, Value, Tags with custom tags.

Reply:
I thought this was more desirable since the use case is to extend a pre-existing metric.